### DISCUSSION TITLE
Roadmap: Improved TypeScript API

### DISCUSSION THREAD

**Discussion by denolfe** [payloadcms/payload#286] [Roadmap]
## Description

Generate types from collections and further support their usage throughout Payload

## Implementation Detail

- Generate types from collections
- Allow hooks to accept types
- Allow field hooks to accept types
- Fully type local API
- Misc TypeScript improvements

## Potential Breaking Changes

- Updating of types

## Effort
High

**Comment by jmikrut** [comment]
Update: 

We've been making progress on this feature and there is currently a PR open which includes a utility to automatically generate TypeScript interfaces for all of your collections and globals:

https://github.com/payloadcms/payload/pull/363

**Comment by denolfe** [comment]
Update:

PR for better typing of payload operations and hooks has been merged in #368
